src/sentry/snuba/discover.py

```diff
@@ -727,8 +689,59 @@ def timeseries_query(
     if len(snuba_args["aggregations"]) == 1:
         snuba_args["aggregations"][0][2] = "count"
 
-    if conditions is not None:
-        snuba_args["conditions"].extend(conditions)
+    return snuba_args, snuba_filter
+
+
+def key_transaction_timeseries_query(selected_columns, query, params, rollup, referrer, queryset):
+    snuba_args, snuba_filter = get_timeseries_snuba_vars(selected_columns, query, params, rollup)
+    snuba_args.get("conditions").extend(key_transaction_conditions(queryset))
```

Comment:
You may hit a conflict here. I recently merged changes to remove `snuba_args` dictionary from most of the discover internals, as `eventstore.Filter` offers similar features without object attributes.